status/service/api.go

```diff
@@ -25,3 +31,49 @@ func (c *Client) Request(dst *network.ServerIdentity) (*Response, error) {
 	}
 	return resp, nil
 }
+
+// Connectivity sends a message from all nodes to all nodes in the list
+// and checks if all messages are received correctly. If findFaulty == true,
+// then the service will try very hard to get a list of nodes that can
+// communicate with each other.
+//
+// The return value of this call is the set of nodes that can communicate
+// with each other.
+func (c *Client) Connectivity(priv kyber.Scalar, list []*network.ServerIdentity,
+	timeout time.Duration, findFaulty bool) ([]*network.ServerIdentity, error) {
+	conn := &Connectivity{
+		List:       list,
+		Timeout:    int64(timeout),
+		FindFaulty: findFaulty,
+		Time:       time.Now().Unix(),
+	}
+	hash, err := conn.hash()
+	if err != nil {
+		return nil, err
+	}
+	conn.Signature, err = schnorr.Sign(cothority.Suite, priv, hash)
+	if err != nil {
+		return nil, err
+	}
+	resp := &ConnectivityReply{}
+	err = c.SendProtobuf(list[0], conn, resp)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Nodes, nil
+}
+
+func (c *Connectivity) hash() ([]byte, error) {
```

Comment:
This does not include the `Time` value so one could change the value and send the message around. Same for `FindFaulty` but it's less of a big deal.

Reply:
Thanks, missed that. Updated to hash:

```
Time | Timeout | FindFaulty ? byte(1) : byte(0) | List
```